mindsdb/integrations/handlers/autokeras_handler/autokeras_handler.py

```diff
@@ -0,0 +1,151 @@
+
+    training_df = pd.get_dummies(training_df)
+    logger.info("In train(): after getting dummies")
+    logger.info(training_df)
+    logger.info(training_df.columns.values.tolist())
+
+    categorical_dummy_column_names = [col for col in training_df.columns.values.tolist() if col not in numeric_column_names]
+
+    trainer.fit(training_df, df[target], epochs=DEFAULT_EPOCHS)
+    return trainer.export_model(), categorical_dummy_column_names
+
+
+def get_preds_from_autokeras_model(df, model, target, all_column_names):
+    cols_to_drop = ["__mindsdb_row_id", target]
+    for col in cols_to_drop:
+        if col in df.columns.values.tolist():
+            df = df.drop(col, axis=1)
+    logger.info("In get_preds(): before getting dummies")
+    logger.info(df)
+    logger.info(target)
+    logger.info("In get_preds(): after getting dummies")
+    df = pd.get_dummies(df)
+    logger.info(df)
+    logger.info("In get_preds(): before filling missing columns")
+    logger.info(all_column_names)
+    logger.info(df.columns.values.tolist())
+    for col in all_column_names:
+        if col not in df.columns.values.tolist():
+            df[col] = 0
+    logger.info("In get_preds(): after filling missing columns")
+    logger.info(df)
+
+    return model.predict(df)
+
+
+
+class AutokerasHandler(BaseMLEngine):
+    """
+    Integration with the AutoKeras ML library.
+    """  # noqa
+
+    name = 'autokeras'
+
+    def create(self, target: str, df: Optional[pd.DataFrame] = None, args: Optional[dict] = None) -> None:
+        """
+        Create and train model on the input df
+        """
+        args = args['using']  # ignore the rest of the problem definition
+        args["target"] = target
+        args["folder_path"] = "autokeras"
+
+        logger.info("In create(): Before training")
+        logger.info(df)
+        logger.info(df.dtypes)
+        logger.info(df.shape)
+        logger.info(target)
+        args["training_data_column_count"] = len(df.columns) - 1 # subtract 1 for target
+        model, args["data_column_names"] = train_autokeras_model(df, target)
+
+        model.save(args["folder_path"])
+        self.model_storage.json_set("predict_args", args)
+
+    
+    def predict(self, df, args=None):
+        args = self.model_storage.json_get("predict_args")
+        logger.info("Before load model")
+        model = load_model(args["folder_path"], custom_objects=ak.CUSTOM_OBJECTS)
+
+        df_to_predict = df.copy()
+        logger.info(df_to_predict)
+        logger.info(df_to_predict.dtypes)
+        logger.info(df.shape)
+        logger.info("Before get predictions")
+        logger.info(f'predict col len: {len(df_to_predict.columns)} | training col len: {args["training_data_column_count"]}')
+        if len(df_to_predict.columns) != args["training_data_column_count"]:
+            # TODO: rephrase to something more user-friendly
+            raise Exception("All columns must be specified in the WHERE clause of the predict query")
```

Comment:
They don't have to specify all columns in the WHERE query, they only have to specify the columns they'd like to filter by when doing the prediction.

Unless I've missed something?